charts/posthog/values.yaml

```diff
@@ -1852,9 +1852,47 @@ prometheus:
               labels:
                 severity: critical
               annotations:
-                summary: Captured events, including session recording, have been delayed by more than 5 minutes for more than 5 minutes.
+                summary: Ingestion for analytics events has been delayed by more than 5 minutes for more than 5 minutes.
                 description: "Latest timestamp registered as processed on topic 'events_plugin_ingestion', consumer group 'ingestion' more then 5 minutes old"
 
+            - alert: RecordingsConsumerDelayed
+              # TODO: fix these consumer delay alerts. The issue with the
+              # current alerts is that they are not the _oldest_ message not
+              # processed but the _newest_ of the oldest _per partition_. This
+              # applies to all the alerts for consumer lag above. Adjusting the
+              # query to the below could work, but also simply using absolute
+              # lag may be the least work
+              #
+              # ```
+              # (time() * 1000 - (
+              #   min(
+              #     max(
+              #       latest_processed_timestamp_ms{groupId="session-recordings",topic="session_recording_events"}
+              #     ) by (topic)
+              #   )
+              # ))
+              # ```
+              #
+              # The issue with the above is when there is no volume on a
+              # partition, then we end up not reporting a timestamp for this
+              # partition and the alert will sound incorrectly.
+              expr: (time() * 1000 - (max(latest_processed_timestamp_ms{groupId="session-recordings",topic="session_recording_events"}))) > 300000
+              for: 5m
+              labels:
+                severity: critical
+              annotations:
+                summary: Ingestion for session recording and web performance events has been delayed by more than 5 minutes for more than 5 minutes.
+                description: "Latest timestamp registered as processed on topic 'session_recording_events', consumer group 'session-recordings' more then 5 minutes old"
```

Comment:
should there be a runbook linked here or does that go somewhere else?

Reply:
I have added a dashboard [here](https://grafana.us-east-1.prod.posthog.dev/d/CT2FdnoVz/session-recordings?orgId=1&from=now-1h&to=now) for session recordings that an intend to link to from the runbook

@benjackwhite @pauldambra do you have any changes you think would be useful on the dashboard? What do you think is important for session recordings?

Reply:
Dashboard is great.... We'll know what's missing during the next outage 🤣 

Interesting that we get this log a lot "a client request body is buffered to a temporary file"

Might be interesting to see if more RAM on nginx or a larger [client_body_buffer_size](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_body_buffer_size) would increase throughput by avoiding disk

Can we graph how long nginx is taking processing requests?

Reply:
hmmm 🤔 the question we always ask during an incident with session recording is "is this traffic coming from one team or several"? Can we answer that with this dashboard?

Reply:
I'll add the response times 👍 

I tried adding [breakdown by team and failed](https://grafana.us-east-1.prod.posthog.dev/explore?orgId=1&left=%7B%22datasource%22:%22P8E80F9AEF21F6940%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22editorMode%22:%22code%22,%22expr%22:%22count%20%28%7Bcontainer%3D%5C%22posthog-events%5C%22%7D%20%7C%20json%20%7C%20unwrap%20team_id%20%5B1m%5D%29%20by%20%28team_id%29%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D)

I wish my logql was better, I'll leave that for someone else to add 😢

Reply:
@tiina303 I've add https://github.com/PostHog/product-internal/blob/main/infrastructure/runbooks/posthog/session-recordings.md but I'm not too sure about the runbook strategy and would rather get this in as is for now.

Reply:
Grabbed that client_body_buffer_size possible insight here https://github.com/PostHog/posthog/issues/13812

(cc @benjackwhite)